doc/spec/namespace.md

```diff
@@ -0,0 +1,124 @@
+# Namespace
+The namespace represents the full set of named repositories which can be
+referenced by Docker. A repository contains a set of content-addressable blobs
+and tags referencing those blobs. The namespace can be used to discover the
+location and trust information of a repository based on DNS, HTTP requests and
+other means. The namespace is global and owned through the domain at the root of
+the namespace path.
+
+A repository should always be referenced by its fully canonical value. If a
+client presents a shortened name to a user, that name should be fully
+canonicalized based on rules defined by the client before contacting a remote
+server. When mirroring repositories, the original name for a repository must be
+used. Changing a repository name is equivalent to copying or moving and will not
+be referenced by the original repository.
+
+## Format
+A name consists of two parts, a DNS host name plus a repository path. The host
```

Comment:
it would be nice if the 'default' registry deployment was targeted at port 443, with a cert, and reverse proxy (if required).  Then just give instructions to cut out those parts if people don't want them.

I doubt anyone actually wants to run their registry publicly on http/5000.  Its done because for a long time, there wasn't a documented solution to do it properly.

Secure by default is a thing